src/components/panel/panel.js

```diff
@@ -527,5 +531,16 @@ MdPanelRef.prototype._createPanel = function() {
         if (self._config['panelClass']) {
           self._panelEl.addClass(self._config['panelClass']);
         }
+
+        self._addStyles();
       });
 };
+
+
+/**
+ * Adds the styles for the panel, such as positioning and z-index.
+ * @private
+ */
+MdPanelRef.prototype._addStyles = function() {
+  this._panelContainer.css('z-index', this._config['zIndex'] || defaultZIndex);
```

Comment:
I set the default z-index in the default config object. This way, if the user sets z-index to 0, it will overwrite the default value and I don't need to do the checks here. Added a test for this case as well.